src/argilla/utils/modeling.py

```diff
@@ -0,0 +1,257 @@
+from argilla.client.feedback.dataset.remote.dataset import RemoteFeedbackDataset
+from argilla.client.feedback.schemas.metadata import (
+    FloatMetadataProperty,
+    IntegerMetadataProperty,
+    TermsMetadataProperty,
+)
+from argilla.client.feedback.schemas.records import FeedbackRecord
+from argilla.client.feedback.schemas.remote.records import RemoteFeedbackRecord
+
+
+# %%
+class TextDescriptivesExtractor:
+    """This class extracts a number of basic text descriptives from FeedbackDataset
+    records using the TextDescriptives library and adds them as record metadata."""
+
+    def __init__(
+        self,
+        model: str = "en",
+        metrics: Optional[List[str]] = None,
+        fields: Optional[List[str]] = None,
+        visible_for_annotators: bool = True,
+        progress: bool = True,
+    ):
+        """
+        Initialize a new TextDescriptivesExtractor object.
+
+        Args:
+            model (str): The language model to use for text descriptives.
+            metrics (Optional[List[str]]): A list of metrics to extract. If None, all metrics will be extracted.
+            fields (Optional[List[str]]): A list of field names to extract metrics from. If None, all fields will be used.
+            visible_for_annotators (bool): Whether the extracted metrics should be visible to annotators.
+            progress (bool): Whether to show a progress bar when extracting metrics.
+        """
+        self.model = model
+        self.metrics = metrics
+        self.fields = fields
+        self.visible_for_annotators = visible_for_annotators
+        self.progress = progress
+        self.__basic_metrics = [
+            "n_tokens",
+            "n_unique_tokens",
+            "n_sentences",
+            "perplexity",
+            "entropy",
+            "flesch_reading_ease",
+        ]
+
+    def _extract_metrics_for_single_field(
+        self,
+        records: List[Union[FeedbackRecord, RemoteFeedbackRecord]],
+        field: str,
+        basic_metrics: Optional[List[str]] = None,
+    ) -> Optional[pd.DataFrame]:
+        """
+        Extract text descriptives metrics for a single field from a list of feedback records
+        using the TextDescriptives library.
+
+        Args:
+            records (List[Union[FeedbackRecord, RemoteFeedbackRecord]]): A list of FeedbackDataset or RemoteFeedbackDataset records.
+            field (str): The name of the field to extract metrics for.
+            basic_metrics (Optional[List[str]]): A list of basic metrics to extract. If None, all metrics will be extracted.
+
+        Returns:
+            Optional[pd.DataFrame]: A dataframe containing the text descriptives metrics for the field, or None if the field is empty.
+        """
+        # If the field is empty, skip it
+        if not any(record.fields[field] for record in records):
+            return None
+        field_text = [record.fields[field] for record in records]
```

Comment:
```suggestion
        # If the field is empty, skip it
        field_text = [record.fields[field] for record in records if record.fields.get(field, None)]
        if len(field_text) < 1:
            return None
```

Reply:
Iterating just once over the records is preferred, since for the `RemoteFeedbackDataset` those will be lazily fetched, and in consequence, fetched twice in this function call

Reply:
Totally makes sense. I modified your suggestion to avoid comparing a list and integer and avoid `field_metrics` being referenced before assignment:

```python
      # If the field is empty, skip it
      field_text = [record.fields[field] for record in records if record.fields[field]]
      if not field_text:
          return None
      # If basic metrics is None, use all basic metrics
      field_metrics = pd.DataFrame()
```